org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/preferences/SonarLintProjectConfigurationManager.java

```diff
@@ -96,6 +97,16 @@ public SonarLintProjectConfiguration load(IScopeContext projectScope, String pro
     }
     projectConfig.setAutoEnabled(projectNode.getBoolean(P_AUTO_ENABLED_KEY, true));
     projectConfig.setBindingSuggestionsDisabled(projectNode.getBoolean(P_BINDING_SUGGESTIONS_DISABLED_KEY, false));
+
+    // When importing a project (but not when restarting the workspace), the when accessing the project preferences for
+    // the first time, they cannot be loaded and will fallback to the default value (in this case here "true"). This
+    // seems to be coming from Eclipse itself as the preferences are loaded lazily. THis is no problem as when a
+    // project is imported right after importing all files will be shown as changed (added) and then the preference
+    // will be read again and then it is available!
+    // When importing a project, the preferences are not loaded lazily anymore but fetched from cache and are not
+    // falling back to the default values!
```

Comment:
Nitpick: there are a few typos in this comment - I can fix the typos in my mind and understand now in the middle of the review, but it may be harder in 6 months 😅